aws-sdk-core/lib/aws-sdk-core/plugins/s3_accelerate.rb

```diff
@@ -0,0 +1,73 @@
+module Aws
+  module Plugins
+
+    # Provides support for using `Aws::S3::Client` with Amazon S3 Transfer
+    # Acceleration.
+    #
+    # Go here for more information about transfer acceleration:
+    # [http://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html](http://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html)
+    #
+    # @seahorse.client.option [Boolean] :use_accelerate_endpoint (false)
+    #   When set to `true`, accelerated bucket endpoints will be used
+    #   for all object operations. You must first enable
+    #   accelerate for each bucket. [Go here for more information](http://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html).
+    #
+    class S3Accelerate < Seahorse::Client::Plugin
+
+      option(:use_accelerate_endpoint, false)
+
+      def add_handlers(handlers, config)
+        operations = config.api.operation_names - [
+          :create_bucket, :list_buckets, :delete_bucket,
```

Comment:
Completely understood, not sure of a valid alternative. `#create_bucket` and `#list_buckets` are simply not supported. The reason is the s3-accelerate endpoint must have an accelerated bucket as the endpoint prefix. Create is creating a new bucket that does not previously exist and list buckets is outside the context of a single bucket. `#delete_bucket` was on request from the S3 team. There is no deterministic characteristic of these that makes it possible to not hardcode.